modules/core/src/lib/attribute-manager.js

```diff
@@ -333,6 +342,29 @@ export default class AttributeManager {
   }
   /* eslint-enable max-statements */
 
+  _addShaderAttributes(attribute, shaderAttributes, extraProps) {
+    attribute.userData.shaderAttributes = {};
+
+    for (const shaderAttributeName in shaderAttributes) {
+      const shaderAttribute = shaderAttributes[shaderAttributeName];
+
+      // Initialize the attribute descriptor, with WebGL and metadata fields
+      attribute.userData.shaderAttributes[shaderAttributeName] = new Attribute(
+        this.gl,
+        Object.assign({}, shaderAttribute, {
```

Comment:
Any way we can reuse this attribute descriptor -> Attribute instance logic?

Reply:
Done. Created a `_createAttribute` method that can handle both cases.